### PR TITLE
xdg-desktop-portal: update to 1.8.0

### PR DESCRIPTION
**Pull request by concatime**

### CODE REVIEW

**Comment by q66** [comment]
should be able to enable libportal now

**Comment by concatime** [comment]
@Duncaen, are you okay with this patch?
I enabled libportal as @q66 told me, and removed `pre_configure` hook.

```diff
diff --git a/srcpkgs/xdg-desktop-portal/template b/srcpkgs/xdg-desktop-portal/template
index a25c486aad..11908fbdd3 100644
--- a/srcpkgs/xdg-desktop-portal/template
+++ b/srcpkgs/xdg-desktop-portal/template
@@ -3,9 +3,10 @@ pkgname=xdg-desktop-portal
 version=1.8.0
 revision=1
 build_style=gnu-configure
-configure_args="--enable-pipewire --enable-geoclue --disable-libportal"
-hostmakedepends="automake libtool gettext-devel pkg-config glib-devel"
-makedepends="flatpak-devel fuse-devel pipewire-devel geoclue2-devel"
+configure_args="--enable-pipewire --enable-geoclue --enable-libportal"
+hostmakedepends="pkg-config"
+makedepends="flatpak-devel fuse-devel pipewire-devel geoclue2-devel
+ libportal-devel"
 short_desc="Portal frontend service for Flatpak"
 maintainer="Duncaen <duncaen@voidlinux.org>"
 license="LGPL-2.1-or-later"
@@ -13,10 +14,6 @@ homepage="https://github.com/flatpak/xdg-desktop-portal"
 distfiles="https://github.com/flatpak/${pkgname}/releases/download/${version}/${pkgname}-${version}.tar.xz"
 checksum=a2fc5681b3a35078239072a98d6435a4c8404016730cab17c9febfd4ecec3610
 
-pre_configure() {
-	autoreconf -vfi
-}
-
 post_install() {
 	rm -fR "${DESTDIR}/usr/lib/systemd"
 }
```

**Comment by Duncaen** [comment]
I just happen to package it initially, I don't use or care about it in any way. Its an OK from me.

File: srcpkgs/xdg-desktop-portal/template [hunk 1]
```diff
@@ -11,12 +11,12 @@ maintainer="Duncaen <duncaen@voidlinux.org>"
...
-checksum=883c9c9a925e48af54812b5347f546dd776ec2e27076a83d5a8126af6bafb9da
+checksum=a2fc5681b3a35078239072a98d6435a4c8404016730cab17c9febfd4ecec3610
 
 pre_configure() {
 	autoreconf -vfi
 }
 
 post_install() {
-	rm -rf "${DESTDIR}/usr/lib/systemd"
+	rm -fR "${DESTDIR}/usr/lib/systemd"
```
Comment (Duncaen) [inline srcpkgs/xdg-desktop-portal/template:21]: Is there a reason to change that?

Comment (concatime) [reply srcpkgs/xdg-desktop-portal/template:18]: Well, technically, `-R` is the official argument for recursion. `-r` is just an alias.
If you look at other commands, like `umount`, `-R` is for recursion.

Comment (concatime) [reply srcpkgs/xdg-desktop-portal/template:18]: > -R
> Remove file hierarchies. See the DESCRIPTION.
> -r
> Equivalent to -R.

https://pubs.opengroup.org/onlinepubs/9699919799/utilities/rm.html

**Comment by q66** [comment]
make sure to try tests too (the check target)

File: srcpkgs/xdg-desktop-portal/template (hunk 1 above)
Comment (ahesford) [reply srcpkgs/xdg-desktop-portal/template:18]: If you're going to be pedantic, I'll pedantically note that "equivalence" and "alias" are two different concepts. There is no merit in this change.

Comment (concatime) [reply srcpkgs/xdg-desktop-portal/template:18]: > -R
> Copy file hierarchies.

https://pubs.opengroup.org/onlinepubs/9699919799/utilities/cp.html

Comment (ahesford) [reply srcpkgs/xdg-desktop-portal/template:18]: What `cp` and `umount` do for recurrence are of no consequence. This is an `rm` invocation. The `rm` man page, from your first link, establishes that `-r` and `-R` are *equivalent*. There is no "official" argument and one is not an "alias". Equivalence means that the two alternatives can be used interchangably *in every instance*.

Stop trying to argue for this change on principle. This is a purely stylistic change with no tangible benefit that has the drawback of concealing the functional history of this line in the template. It is without merit.

**Comment by q66** [comment]
do the tests pass? if so, squash the commits, and this is ready to go

**Comment by concatime** [comment]
```
ERROR: test-portals - Bail out! GLib-GIO:ERROR:../gio/gtestdbus.c:622:start_daemon: assertion failed (error == NULL): Failed to spawn child process ?dbus-daemon? (No such file or directory) (g-exec-error-quark, 19)
PASS: testdb 1 /db/open
PASS: testdb 2 /db/serialize
PASS: testdb 3 /db/modify
SKIP: test-doc-portal 1 /db/version # SKIP fuse_mount: Operation not permitted
SKIP: test-doc-portal 2 /db/create_doc # SKIP fuse_mount: Operation not permitted
SKIP: test-doc-portal 3 /db/recursive_doc # SKIP fuse_mount: Operation not permitted
SKIP: test-doc-portal 4 /db/create_docs # SKIP fuse_mount: Operation not permitted
SKIP: test-doc-portal 5 /db/add_named # SKIP fuse_mount: Operation not permitted
ERROR: test-permission-store - Bail out! GLib-GIO:ERROR:../gio/gtestdbus.c:622:start_daemon: assertion failed (error == NULL): Failed to spawn child process ?dbus-daemon? (No such file or directory) (g-exec-error-quark, 19)
SKIP: tests/test-document-fuse.sh - No cap_sys_admin in bounding set, can't use FUSE
============================================================================
Testsuite summary for xdg-desktop-portal 1.8.0
============================================================================
# TOTAL: 11
# PASS:  3
# SKIP:  6
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 2
```

**Comment by concatime** [comment]
```
PASS: test-portals 1 /portal/account/exists
PASS: test-portals 2 /portal/account/basic
PASS: test-portals 3 /portal/account/delay
PASS: test-portals 4 /portal/account/cancel
PASS: test-portals 5 /portal/account/close
PASS: test-portals 6 /portal/account/reason
PASS: test-portals 7 /portal/account/parallel
PASS: test-portals 8 /portal/background/exists
PASS: test-portals 9 /portal/background/basic1
PASS: test-portals 10 /portal/background/basic2
PASS: test-portals 11 /portal/background/commandline
PASS: test-portals 12 /portal/background/reason
PASS: test-portals 13 /portal/camera/exists
PASS: test-portals 14 /portal/camera/basic
PASS: test-portals 15 /portal/camera/delay
PASS: test-portals 16 /portal/camera/close
PASS: test-portals 17 /portal/camera/cancel
PASS: test-portals 18 /portal/camera/lockdown
PASS: test-portals 19 /portal/camera/noaccess1
PASS: test-portals 20 /portal/camera/noaccess2
PASS: test-portals 21 /portal/camera/parallel
PASS: test-portals 22 /portal/email/exists
PASS: test-portals 23 /portal/email/basic
PASS: test-portals 24 /portal/email/delay
PASS: test-portals 25 /portal/email/cancel
PASS: test-portals 26 /portal/email/close
PASS: test-portals 27 /portal/email/address
PASS: test-portals 28 /portal/email/subject
PASS: test-portals 29 /portal/email/parallel
PASS: test-portals 30 /portal/filechooser/exists
PASS: test-portals 31 /portal/gamemode/exists
PASS: test-portals 32 /portal/inhibit/exists
PASS: test-portals 33 /portal/inhibit/basic
PASS: test-portals 34 /portal/inhibit/delay
PASS: test-portals 35 /portal/inhibit/close
PASS: test-portals 36 /portal/inhibit/cancel
PASS: test-portals 37 /portal/inhibit/parallel
PASS: test-portals 38 /portal/inhibit/permissions
ERROR: test-portals - Bail out! xdg-desktop-portal:ERROR:tests/inhibit.c:327:session_state_changed_cb: assertion failed (state == XDP_LOGIN_SESSION_RUNNING): (2 == 1)
PASS: testdb 1 /db/open
PASS: testdb 2 /db/serialize
PASS: testdb 3 /db/modify
SKIP: test-doc-portal 1 /db/version # SKIP fuse_mount: Operation not permitted
SKIP: test-doc-portal 2 /db/create_doc # SKIP fuse_mount: Operation not permitted
SKIP: test-doc-portal 3 /db/recursive_doc # SKIP fuse_mount: Operation not permitted
SKIP: test-doc-portal 4 /db/create_docs # SKIP fuse_mount: Operation not permitted
SKIP: test-doc-portal 5 /db/add_named # SKIP fuse_mount: Operation not permitted
PASS: test-permission-store 1 /permissions/version
PASS: test-permission-store 2 /permissions/change
PASS: test-permission-store 3 /permissions/lookup
PASS: test-permission-store 4 /permissions/delete1
PASS: test-permission-store 5 /permissions/delete2
PASS: test-permission-store 6 /permissions/delete3
PASS: test-permission-store 7 /permissions/delete4
PASS: test-permission-store 8 /permissions/create1
PASS: test-permission-store 9 /permissions/create2
PASS: test-permission-store 10 /permissions/set-value
SKIP: tests/test-document-fuse.sh - No cap_sys_admin in bounding set, can't use FUSE
```

**Comment by ericonr** [comment]
> checkdepends="dbus" should fix this.

If it doesn't, run the tests with `dbus-run-session <tests>`

**Comment by concatime** [comment]
@ericonr
```
dbus[6734]: Could not get password database information for UID of current process: User "???" unknown or no memory to allocate password entry

dbus[6734]: Failed to start message bus: Memory allocation failure in message bus
dbus-run-session: EOF reading address from bus daemon
```

**Comment by ericonr** [comment]
hmm no idea what could have caused that :o

**Comment by q66** [comment]
if a test fails because of environment restrictions, you can always patch out the test from running, if it's not too complicated (should be very easy with meson, just comment out the appropriate lines)

if it is complicated, then just do

```
# reason here
do_check() {
	:
}
```

**Comment by q66** [comment]
running tests but ignoring them is not good because it defeats the point of running tests at all - i.e. to ensure that the ones you expect to pass actually pass

**Comment by Duncaen** [comment]
Merged in https://github.com/void-linux/void-packages/commit/ef8cc6480d416fd36528648e2712b258d79d7087.

**Pull request closed without merging**